src/libraries/System.ObjectModel/tests/TrimmingTests/TypeConverterAttributeCtorTest.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+
+namespace TypeConverterAttributeTest
+{
+    /// <summary>
+    /// Tests that the public constructors of types passed into System.ComponentModel.TypeConverterAttribute
+    /// are not trimmed out when needed in a trimmed application.
+    /// </summary>
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            // Type-based TypeConverterAttribute ctor overload, ensure public parameterized ctor of TypeConverter type is preserved.
+            TypeDescriptor.AddAttributes(typeof(DayOfWeek), new TypeConverterAttribute(typeof(EnumConverter)));
```

Comment:
I'd be careful using a built-in type `EnumConverter` here. There will be other reasons this type gets preserved correctly - see https://github.com/dotnet/runtime/issues/38582.

We should probably be using a custom type to ensure this test is testing the overload correctly.